src/lustre/lustreAstInlineConstants.ml

```diff
@@ -275,6 +275,7 @@ and push_pre is_guarded pos =
   | ArrayConcat (p, e1, e2) -> ArrayConcat (p, r e1, r e2)
   | ArraySlice (p, e1, (e2, e3)) -> ArraySlice (p, r e1, (e2, e3))
   | Quantifier (p, e1, l, e2) -> Quantifier (p, e1, l, r e2)
+  | ChooseOp (p, i, e) -> ChooseOp (p, i, r e)
```

Comment:
This code should be unreachable since choose operators are desugared before this point. Replace it with `assert false` and comment.

Reply:
addressed